Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h

```diff
@@ -72,54 +84,54 @@ namespace AZ
             {
                 if constexpr (AZStd::is_integral_v<DestType> && !AZStd::is_same_v<DestType, bool>)
                 {
-                    if (AttributeRead<bool, DestType, Args...>(value, attr, instance, args ...))
+                    if (AttributeRead<bool, DestType, Args...>(value, attr, AZStd::forward<InstType>(instance), args...))
```

Comment:
Arguments should not be forwarded.
The parameters are `const&` here.

Reply:
You know what, I see don't a reason why the arugments here need to be a `const&` reference since only once of the Attribute functions calls will eventually be invoked.

Reply:
Actually I realize the problem now.
The `AttributeFunction` types have an specific signature.
i.e `AttributeFunction<int(void*, AZStd::string)>`

The problem is forwarding the arguments pass the reference type for the arguments to the AttributeRead function, where the cast would fail.
i.e `AttributeFunction<int(void*, AZStd::string&)>` can't cast an `AttributeFunction<int(void*, AZStd::string)>` with the **&** as they are two completely unrelated types